scripts/build/build_libpq.sh

```diff
@@ -12,30 +12,69 @@ postgres_version="${LIBPQ_VERSION}"
 # last release: https://www.openssl.org/source/
 openssl_version="${OPENSSL_VERSION}"
 
+# last release: https://kerberos.org/dist/
+krb5_version="1.21.3"
+
+# last release: https://www.gnu.org/software/gettext/
+gettext_version="0.22.5"
+
 # last release: https://openldap.org/software/download/
-ldap_version="2.6.3"
+ldap_version="2.6.8"
 
 # last release: https://github.com/cyrusimap/cyrus-sasl/releases
 sasl_version="2.1.28"
 
 export LIBPQ_BUILD_PREFIX=${LIBPQ_BUILD_PREFIX:-/tmp/libpq.build}
 
-if [[ -f "${LIBPQ_BUILD_PREFIX}/lib/libpq.so" ]]; then
+case "$(uname)" in
+    Darwin)
+        ID=macos
+        library_suffix=dylib
+        ;;
+
+    Linux)
+        source /etc/os-release
+        library_suffix=so
+        ;;
+
+    *)
+        echo "$0: unexpected Operating system: '$(uname)'" >&2
+        exit 1
+        ;;
+esac
+
+if [[ -f "${LIBPQ_BUILD_PREFIX}/lib/libpq.${library_suffix}" ]]; then
     echo "libpq already available: build skipped" >&2
     exit 0
 fi
 
-source /etc/os-release
-
 case "$ID" in
     centos)
         yum update -y
         yum install -y zlib-devel krb5-devel pam-devel
+        curl="$(which curl)"
         ;;
 
     alpine)
         apk upgrade
         apk add --no-cache zlib-dev krb5-dev linux-pam-dev openldap-dev openssl-dev
+        curl="$(which curl)"
+        ;;
+
+    macos)
+        brew install automake m4 libtool
+        # If available, libpq seemingly insists on linking against homebrew's
+        # openssl no matter what so remove it. Since homebrew's curl depends on
+        # it, force use of system curl.
```

Comment:
Question: why don't we remove homebrew curl too and just leave the system one?

Reply:
Good point. I'll give that a go.

Reply:
I'm already testing it, no worries :)

Reply:
Looks like it's working ok. https://github.com/psycopg/psycopg2/actions/runs/12622307173